R/utilities.R

```diff
@@ -2731,6 +2906,50 @@ collate_lymphgen = function(sample_table,
 }
 
 
+#' Expand a sample_table (meta data) horizontally with different flavours of lymphgen data.
+#'
+#' @param sample_table Input data frame with sample IDs in the firs column.
+#'
+#' @return
+#' @export
+#' @import tidyverse
+#'
+#' @examples
+#' metadata = get_gambl_metadata()
+#' colalted_meta = collate_lymphgen(sample_table = metadata)
+#' 
+collate_lymphgen = function(sample_table){
+  
+  #setup paths
+  repo_base = config::get("repo_base") #get repo base
+  flavours = config::get("results_merged_wildcards")$lymphgen_template #get all flavours
+  flavours = str_split(flavours, pattern = ",") #seperate each flavour on ","
+  flavours = unlist(flavours) #unlist flavours
+  lymphgen_template = config::get("results_versioned")$lymphgen_template #get tamplate for lymphgen file path
+  lymphgen_path = paste0(repo_base, lymphgen_template) #combine with repo_base path
+  lymphgen_full_paths = glue::glue(lymphgen_path, flavour = flavours) #substitute {flavour} wildcard in template with actual flavours
+  
+  #print paths
+  message(paste0("Reading from the following paths:"))
+  message(paste0(for(i in lymphgen_full_paths) print(i)))
+  
+  #read files into R
+  for (f in lymphgen_full_paths){
```

Comment:
I can think of a few edge cases where this function could break, namely whether any new versions of LymphGen outputs (i.e. hypothetical future files that don't start with `GAMBL_all_the_things`) don't have all of the "flavour" combinations. There's also a ton of useful information in the LymphGen table besides just the classification, and there might be use cases to use those as well. 

This is how I would iterate over this: 
```
load_lymphgen <- function(flavour, lymphgen_path){
    lg_path <- glue::glue(lymphgen_path, flavour = flavour)
    if(!file.exists(lg_path){return()} # Ignore missing flavours
    lg_df <- read_tsv(lg_path) %>% 
        mutate(flavour = flavour) # Append the flavour in its own column called "flavour"
    return(lg_df)
}
lymphgen_results <- lapply(flavours, load_lymphgen, lymphgen_path = lymphgen_path)
lymphgen_results <- bind_rows(lymphgen_paths) # Now you have all the lymphgen results tables stacked on top of each other, with the results from each flavour identified by the `flavour` column
lymphgen_results <- tidy_lymphgen(lymphgen_results, lymphgen_column_in = "Subtype.Prediction", lymphgen_column_out = "LymphGen") 
if(!tidy){ 
    # "Tidy" results are usually long format, so the wide format is the "untidy" version
    lymphgen_untidy <- lymphgen_results %>% 
        select(Sample.Name, Subtype.Prediction, LymphGen, flavour) %>% 
        # This keeps both the original Subtype.Prediction and tidied LymphGen values and puts the values from each "flavour" in its own column
        pivot_wider(
            names_from = flavour, 
            values_from = c(Subtype.Prediction, LymphGen), 
            names_glue = c({.value}_{flavour})
        )
    # Do some stuff to join this with the metadata (if provided) or subset to the sample_ids (if provided)
    ...
    return(lymphgen_untidy)
} else {
    # Join with metadata or subset to sample_ids
    return(lymphgen_tidy)
    # Notice I didn't drop any columns from this version. There's lots of information in the "tidy" table about feature_count, list of features used in classification, classification confidence, etc. that can be kept in the tidy version. 
}

```

Reply:
Great suggestion, I have incorporated parts of this into the updated function, thanks @lkhilton